api/osf_api.py

```diff
@@ -784,3 +784,51 @@ def get_user_developer_app_data(session, app_id=None):
     if data:
         return data
     return None
+
+
+def create_personal_access_token(
+    session, name='OSF Test PAT', scopes='osf.nodes.full_read'
+):
+    """Create a Personal Access Token for the user that is currently logged in to OSF
+    (via session object). Default scope is 'osf.nodes.full_read'. The scopes parameter
+    is a string (not a list) with multiple scopes separated by a space.
+    EX: 'osf.users.profile_write osf.full_write osf.nodes.full_write'
+    """
+    if not session:
```

Comment:
Unrelated to CR but a general selenium question: wondering what will happen when `session` is `None` and the default session is used. In our code, we always pass session to the function when calling `create_personal_access_token()`.

Reply:
Here's the code for get_default_session():

def get_default_session():
    return client.Session(
        api_base_url=settings.API_DOMAIN,
        auth=(settings.USER_ONE, settings.USER_ONE_PASSWORD),
    )

We’re basically just using the credentials for our User One (which is the primary user we are always signed in as) and we create a session object for that user’s credentials.  Typically this isn’t necessary since we usually are passing in the session object, but we always include this if session condition in our api functions just in case.